src/app/utils/memory.ts

```diff
@@ -4,147 +4,92 @@ import { PineconeClient } from "@pinecone-database/pinecone";
   private static instance: MemoryManager;
-  private history: Redis;
-  private vectorDBClient: PineconeClient | SupabaseClient;
+
+  private dbClient: SupabaseClient;
 
   public constructor() {
-    this.history = Redis.fromEnv();
-    if (process.env.VECTOR_DB === "pinecone") {
-      this.vectorDBClient = new PineconeClient();
-    } else {
-      const auth = {
-        detectSessionInUrl: false,
-        persistSession: false,
-        autoRefreshToken: false,
-      };
-      const url = process.env.SUPABASE_URL!;
-      const privateKey = process.env.SUPABASE_PRIVATE_KEY!;
-      this.vectorDBClient = createClient(url, privateKey, { auth });
-    }
+    const auth = {
+      detectSessionInUrl: false,
+      persistSession: false,
+      autoRefreshToken: false,
+    };
+    const url = process.env.SUPABASE_URL!;
+    const privateKey = process.env.SUPABASE_PRIVATE_KEY!;
+    this.dbClient = createClient(url, privateKey, { auth });
   }
 
-  public async init() {
-    if (this.vectorDBClient instanceof PineconeClient) {
-      await this.vectorDBClient.init({
-        apiKey: process.env.PINECONE_API_KEY!,
-        environment: process.env.PINECONE_ENVIRONMENT!,
-      });
-    }
-  }
+  public async init() {}
 
-  public async vectorSearch(
-    recentChatHistory: string,
-    companionFileName: string
-  ) {
-    if (process.env.VECTOR_DB === "pinecone") {
-      console.log("INFO: using Pinecone for vector search.");
-      const pineconeClient = <PineconeClient>this.vectorDBClient;
-
-      const pineconeIndex = pineconeClient.Index(
-        process.env.PINECONE_INDEX! || ""
-      );
-
-      const vectorStore = await PineconeStore.fromExistingIndex(
-        new OpenAIEmbeddings({ openAIApiKey: process.env.OPENAI_API_KEY }),
-        { pineconeIndex }
-      );
-
-      const similarDocs = await vectorStore
-        .similaritySearch(recentChatHistory, 3, { fileName: companionFileName })
-        .catch((err) => {
-          console.log("WARNING: failed to get vector search results.", err);
-        });
-      return similarDocs;
-    } else {
-      console.log("INFO: using Supabase for vector search.");
-      const supabaseClient = <SupabaseClient>this.vectorDBClient;
-      const vectorStore = await SupabaseVectorStore.fromExistingIndex(
-        new OpenAIEmbeddings({ openAIApiKey: process.env.OPENAI_API_KEY }),
-        {
-          client: supabaseClient,
-          tableName: "documents",
-          queryName: "match_documents",
-        }
-      );
-      const similarDocs = await vectorStore
-        .similaritySearch(recentChatHistory, 3)
-        .catch((err) => {
-          console.log("WARNING: failed to get vector search results.", err);
-        });
-      return similarDocs;
-    }
-  }
+  public async saveToMemory(content: string, metadata: any) {
+    const embedding = await this.generateEmbedding(content);
+    const supabaseClient = <SupabaseClient>this.dbClient;
+    const error = await supabaseClient
```

Comment:
**issue:** Missing error handling in saveToMemory

The error response from Supabase should be properly handled. Consider throwing an error or returning a result object that indicates success/failure.